plugin/src/main/java/org/opensearch/ml/action/handler/MLSearchHandler.java

```diff
@@ -0,0 +1,73 @@
+/*
+ * Copyright OpenSearch Contributors
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+package org.opensearch.ml.action.handler;
+
+import static org.opensearch.common.xcontent.XContentParserUtils.ensureExpectedToken;
+import static org.opensearch.ml.indices.MLIndicesHandler.ML_TASK_INDEX;
+import static org.opensearch.ml.utils.MLNodeUtils.createXContentParserFromRegistry;
+
+import lombok.extern.log4j.Log4j2;
+
+import org.opensearch.action.ActionListener;
+import org.opensearch.action.ActionRequest;
+import org.opensearch.action.get.GetRequest;
+import org.opensearch.action.get.GetResponse;
+import org.opensearch.client.Client;
+import org.opensearch.common.xcontent.NamedXContentRegistry;
+import org.opensearch.common.xcontent.XContentParser;
+import org.opensearch.index.IndexNotFoundException;
+import org.opensearch.ml.common.exception.MLResourceNotFoundException;
+import org.opensearch.ml.common.parameter.MLTask;
+import org.opensearch.ml.common.transport.task.MLTaskGetRequest;
+import org.opensearch.ml.common.transport.task.MLTaskGetResponse;
+
+/**
+ * Handle general get and search request in ml common.
+ */
+@Log4j2
+public class MLSearchHandler {
+    private final Client client;
+    private NamedXContentRegistry xContentRegistry;
+
+    public MLSearchHandler(Client client, NamedXContentRegistry xContentRegistry) {
+        this.client = client;
+        this.xContentRegistry = xContentRegistry;
+    }
+
+    public void getTask(ActionRequest request, ActionListener<MLTaskGetResponse> actionListener) {
+        MLTaskGetRequest mlTaskGetRequest = MLTaskGetRequest.fromActionRequest(request);
+        String taskId = mlTaskGetRequest.getTaskId();
+        GetRequest getRequest = new GetRequest(ML_TASK_INDEX).id(taskId);
+
+        client.get(getRequest, createGetResponseListener(actionListener, taskId));
```

Comment:
How about move this get request out of `MLSearchHandler` and put search ML tasks/models here?

Reply:
Removed this out of the search hander. But the MLSearchHandler is still added as an empty class for future Search use.